src/acquisition/covid_hosp/common/database.py

```diff
@@ -177,7 +177,9 @@ def insert_dataset(self, publication_date, dataframe):
       ak_cols = self.AGGREGATE_KEY_COLS
 
       # restrict data to just the key columns and remove duplicate rows
-      ak_data = dataframe[ak_cols].drop_duplicates()
+      ak_data = (dataframe[set(ak_cols + self.KEY_COLS)]
+                 .sort_values(self.KEY_COLS)[ak_cols]
+                 .drop_duplicates())
```

Comment:
very nice, so any later-dated rows for the same PK will go into the INSERT...ON DUPLICATE after the earlier ones, and thus the newest data will be preserved in the `ak_table`!  it might be worth leaving a note to that effect?  also (im beginning to hate this data set) should we apply the dataframe_typemap before this sort, because `'2020-10-30'<'2020/09/07'` but not `'20201030'<'20200907'`?

Reply:
> should we apply the dataframe_typemap before this sort

Good call. We've never seen dates without delimiters in the source data, but we _have_ seen both slashes and dashes, and
```
>>> "2/3/4" > "2-3-5"
True
>>> "2/3/4" > "2/3/5"
False
```
😩